### ISSUE TITLE
Add ladder features to the neural network

### ISSUE THREAD

**Issue by Chicoryn** [Chicoryn/dream-go#12]
Add two additional features to the neural network:

* Whether a move at this point is a successful ladder capture.
* Whether a move at this point is a successful ladder escape.

The motivation for adding these two features is that the neural network sometimes fails to read out ladders successfully and reading out a ladder can require the network to look a huge distance (which is not only hard, but would also take up a large amount of features in the neural network).

To help with these problems we can just tell the network directly whether a ladder would work, this should help the network tell whether playing out a ladder is a good idea. There is still an issue with the training data being heavily biased toward avoiding broken ladders (since no human would play them out), which may cause these features to have limited effect. Not sure how to fix that issue except just playing more games.

**Comment by Chicoryn** [comment]
To be able to use DP4A #8  (whenever that gets finished) the number of features must be a multiple of four. So we either need to add two more features or add padding features.

If we are not adding padding features then we can re-add the _scoring_ features. For example, the following two:

* Point is only reachable from our stones.
* Point is only reachable from opponent stones.

These features confirm to the [Tromp-Taylor Rules](https://senseis.xmp.net/?TrompTaylorRules). But we removed some similar features earlier because we disliked them for giving the neural network an overly simplistic view of the game state, so re-adding them may be a bad idea.

**Comment by Chicoryn** [comment]
I added the ladder features and trained another network from human games based on it, and the network still fails the ladder tests.

So questionable whether they are worth the effort, but I am leaving them in and we can see in the future whether they are necessary.
